### PR TITLE
Several fixes across builders and removed some unneeded code

### PR DESCRIPTION
**Pull request by mchf**

### CODE REVIEW

File: src/lib/y2network/interface_config_builders/ib.rb [hunk 1]
```diff
@@ -13,13 +13,17 @@ def initialize
...
+        else
+          @config["IPOIB_MODE"]
+        end
       end
 
       def save
         super
 
-        Yast::LanItems.ipoib_mode = ipoib_mode == "default" ? nil : ipoib_mode
+        @config["IPOIB_MODE"] = ipoib_mode == "default" ? nil : ipoib_mode
```
Comment (jreidinger) [inline src/lib/y2network/interface_config_builders/ib.rb]: do you test this with real write? I sometimes found that transition from LanItems to NetworkInterfaces has many cave eats.

Comment (mchf) [reply src/lib/y2network/interface_config_builders/ib.rb]: yes ... but testing even more just now

Comment (mchf) [reply src/lib/y2network/interface_config_builders/ib.rb]: manually tested and seems to be ok

File: src/lib/y2network/interface_config_builders/dummy.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+require "y2network/interface_config_builder"
+
+module Y2Network
+  module InterfaceConfigBuilders
+    class Dummy < InterfaceConfigBuilder
+      def initialize
+        super(type: "dummy")
+      end
+
+      def save
```
Comment (teclator) [inline src/lib/y2network/interface_config_builders/dummy.rb]: In general miss documentation in `InterfaceConfigBuilder`methods, in particular do not expected to return sometimes the set of an attribute and sometimes `nil`.. specially when more than one attribute is responsibility of a specialized class.

File: src/lib/y2network/interface_config_builders/ib.rb (hunk 1 above)
Comment (teclator) [reply src/lib/y2network/interface_config_builders/ib.rb]: Same than above, please try to document as much as possible the new classes / methods.

File: test/y2network/interface_config_builders/ib_test.rb [hunk 3]
```diff
@@ -27,14 +27,12 @@
 
     it "stores ipoib configuration" do
       subject.ipoib_mode = "datagram"
-      subject.save
-      expect(Yast::LanItems.ipoib_mode).to eq "datagram"
+      expect(subject.save).to eq "datagram"
```
Comment (teclator) [inline test/y2network/interface_config_builders/ib_test.rb]: I would prefert to check the resultant config than the return of the save method

File: test/y2network/interface_config_builders/ib_test.rb [hunk 4]
```diff
@@ -27,14 +27,12 @@
...
     end
 
     it "stores nil to ipoib configuration if mode is 'default'" do
       subject.ipoib_mode = "default"
-      subject.save
-      expect(Yast::LanItems.ipoib_mode).to eq nil
+      expect(subject.save).to eq nil
```
Comment (teclator) [inline test/y2network/interface_config_builders/ib_test.rb]: I would expect to return something like true or false.

**Comment by teclator** [comment]
LGTM

**Pull request merged**